mongomock/aggregate.py

```diff
@@ -488,6 +490,39 @@ def _handle_type_convertion_operator(self, operator, values):
                 'You need to import the pymongo library to support decimal128 type.'
             )
 
+        # Document: https://docs.mongodb.com/manual/reference/operator/aggregation/toDecimal/
+        if operator == '$toDecimal':
+            if not decimal_support:
+                raise NotImplementedError(
+                    'You need to import the pymongo library to support decimal128 type.'
+                )
+            parsed = self.parse(values)
+            if isinstance(parsed, bool):
+                parsed = '1' if parsed is True else '0'
+                decimal_value = decimal128.Decimal128(parsed)
+            elif isinstance(parsed, int):
+                decimal_value = decimal128.Decimal128(str(parsed))
+            elif isinstance(parsed, float):
+                exp = decimal.Decimal('.00000000000000')
+                decimal_value = decimal.Decimal(str(parsed)).quantize(exp)
+                decimal_value = decimal128.Decimal128(decimal_value)
+            elif isinstance(parsed, decimal128.Decimal128):
+                decimal_value = parsed
+            elif isinstance(parsed, str):
+                try:
+                    decimal_value = decimal128.Decimal128(parsed)
+                except decimal.InvalidOperation:
+                    raise OperationFailure(
+                        "Failed to parse number '%s' in $convert with no onError value:"
+                        'Failed to parse string to decimal' % parsed)
+            elif isinstance(parsed, datetime.datetime):
+                epoch = datetime.datetime.utcfromtimestamp(0)
+                string_micro_seconds = str((parsed - epoch).total_seconds() * 1000).split('.')[0]
+                decimal_value = decimal128.Decimal128(string_micro_seconds)
+            else:
+                raise TypeError(" '%s' type is not supported" % type(parsed))
```

Comment:
It seems that you have an unneeded blank at the beginning of the error message, please remove.

Reply:
OK